### DISCUSSION TITLE
The performance of version 7 is much worse than version 6

### DISCUSSION THREAD

**Discussion by j-xzy** [react-hook-form/react-hook-form#5663] [General]
**Describe the bug**
 setValue will trigger much Subject.next in useController and useWatch.

v7:
![image](https://user-images.githubusercontent.com/22634735/122420925-9b3af400-cfbe-11eb-818e-84ca080b57f7.png)

v6.15.4:
![image](https://user-images.githubusercontent.com/22634735/122424086-04bc0200-cfc1-11eb-9122-7011ed5b8c80.png)

**Codesandbox link (Required)**
6.15.5: https://codesandbox.io/s/empty-darkness-muuch?file=/src/App.js
7.8.8 https://codesandbox.io/s/epic-leavitt-pq4ux

**Comment by bluebill1049** [comment]
can i have a v6 version of this? happy to investigate for a better perf solution.

**Reply by j-xzy** [reply]
6.15.5: https://codesandbox.io/s/empty-darkness-muuch?file=/src/App.js
7.8.8 https://codesandbox.io/s/epic-leavitt-pq4ux

maybe too many observers cause performance degradation
